File: app/src/main/java/com/jonasgerdes/stoppelmap/usecases/transportation/station_detail/DepartureDayFragment.java
package com.jonasgerdes.stoppelmap.usecases.transportation.station_detail;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jonasgerdes.stoppelmap.R;
import com.jonasgerdes.stoppelmap.StoppelMapApp;
import com.jonasgerdes.stoppelmap.model.transportation.DepartureDay;
import com.jonasgerdes.stoppelmap.model.transportation.Station;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by Jonas on 15.07.2016.
 */
public class DepartureDayFragment extends Fragment {

    private static final String ARGUMENT_STATION_UUID = "ARGUMENT_STATION_UUID";
    private static final String ARGUMENT_DAY = "ARGUMENT_DAY";

    private Unbinder mUnbinder;
    private DepatureAdapter mDepatureAdapter;
    @BindView(R.id.depatures)
    RecyclerView mDepatureList;


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.transportation_fragment_depature_day, container, false);
        return view;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mUnbinder = ButterKnife.bind(this, view);
        String stationUuid = getArguments().getString(ARGUMENT_STATION_UUID);
        final int dayId = getArguments().getInt(ARGUMENT_DAY);

        mDepatureAdapter = new DepatureAdapter();
        mDepatureList.setAdapter(mDepatureAdapter);
        mDepatureList.setLayoutManager(new LinearLayoutManager(getContext()));

        Station station = StoppelMapApp.getViaActivity(getActivity()).getRealm()
                .where(Station.class).equalTo("uuid", stationUuid).findFirst();
        DepartureDay day = station.getDays().get(dayId);

        mDepatureAdapter.setDepartures(day.getDepartures());
    }

    @Override
    public void onDestroyView() {
        if (mUnbinder != null) {
            mUnbinder.unbind();
        }
        super.onDestroyView();
    }

    public static DepartureDayFragment newInstance(Station station, @DepartureDay.Day int day) {
        Bundle args = new Bundle();
        args.putString(ARGUMENT_STATION_UUID, station.getUuid());
        args.putInt(ARGUMENT_DAY, day);
        DepartureDayFragment fragment = new DepartureDayFragment();
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/usecase/transportation/station_detail/view/departure_day/DepartureDayFragment.kt
package com.jonasgerdes.stoppelmap.usecase.transportation.station_detail.view.departure_day

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.jonasgerdes.stoppelmap.App
import com.jonasgerdes.stoppelmap.R
import com.jonasgerdes.stoppelmap.model.TransportationRepository
import com.jonasgerdes.stoppelmap.model.entity.Station
import kotlinx.android.synthetic.main.transportation_station_detail_departure_day_fragment.*
import javax.inject.Inject

/**
 * @author Jonas Gerdes <dev@jonasgerdes.com>
 * @since 10.08.17
 */
class DepartureDayFragment : Fragment() {

    //todo: refactor in presenter and interactor, there is no time right now...

    @Inject lateinit var repository: TransportationRepository

    init {
        App.graph.inject(this)
    }

    companion object {
        val ARGS_DEPARTURE_DAY = "ARGS_DEPARTURE_DAY"
        val ARGS_STATION_SLUG = "ARGS_STATION_SLUG"

        fun createInstance(station: Station, day: Int) : DepartureDayFragment{
            val args = Bundle()
            args.putString(ARGS_STATION_SLUG, station.uuid)
            args.putInt(ARGS_DEPARTURE_DAY, day)
            val fragment = DepartureDayFragment()
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(
                R.layout.transportation_station_detail_departure_day_fragment,
                container,
                false
        )
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        arguments.getString(ARGS_STATION_SLUG, null)?.let {
            repository.getStationBy(it)?.let {
                depatures.adapter = DepartureAdapter()
                val day = arguments.getInt(ARGS_DEPARTURE_DAY, 0)
                with(depatures.adapter as DepartureAdapter) {
                    departures = it.days!![day].departures
                }
            }
        }
    }
}